### DISCUSSION TITLE
Authentication-OIDC-keycloak

### DISCUSSION THREAD

**Discussion by pokkireddy** [TouK/nussknacker#3995] [Q&A] [answered]
I would like to implement oidc method for authentication, but i am gettting authentication error as "The supplied authentication is not authorized to access this resource"
We are using Keycloak for oidc
Please find the dev-application.conf for authentication details and users.conf file
Attached the network log. you can import the .har file in your chrome to check the request and response. 
Please check all these logs and let me know what is missing.
[oidc.zip](https://github.com/TouK/nussknacker/files/10726010/oidc.zip)

**Comment by gadomsky** [comment, marked as answer]
Hi. Your config `authentication.grant_type` is ignored. You set `authentication.method` = `Oidc`, where it is not possible to confiugre `grant_type` to `password`. `authorization_code` is the only supported value for `grant_type` and is not configurable. Please see possible configuration options [here](https://docs.nussknacker.io/documentation/docs/next/installation_configuration_guide/DesignerConfiguration#openid-connect-security-module)
You can set method to `OAuth2` and then provide configuration as described [here](https://docs.nussknacker.io/documentation/docs/next/installation_configuration_guide/DesignerConfiguration#oauth2-security-module).
 grant_type is then configured under such configpath: `authentication.accessTokenParams.grant_type` (and not `authentication.grant_type` as in your config)
 
Please check out examples  (https://docs.nussknacker.io/documentation/docs/next/installation_configuration_guide/DesignerConfiguration#oauth2-security-module---github-example-with-code-flow and https://docs.nussknacker.io/documentation/docs/next/installation_configuration_guide/DesignerConfiguration#auth0-sample-configuration) in our documentation. I hope it will be helpful. You might also find helpful our [integration test](https://github.com/TouK/nussknacker/blob/staging/security/src/it/scala/pl/touk/nussknacker/ui/security/oauth2/GenericOidcServiceSpec.scala) which uses `KeyCloak` or other tests ([here](https://github.com/TouK/nussknacker/tree/staging/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2) and [here](https://github.com/TouK/nussknacker/tree/staging/security/src/test/scala/pl/touk/nussknacker/ui/security/oidc) )
